pymatgen/entries/compatibility.py

```diff
@@ -648,3 +721,277 @@ def __init__(self, compat_type="Advanced", correct_peroxide=True,
+        print('Mean = ' + str(np.mean(np.array(abs_errors))))
+        print('Std Dev = ' + str(np.std(np.array(abs_errors))))
+        print('Original Error:')
+        print('Median = ' + str(abs(np.median(np.array(self.diffs)))))
+        print('Mean = ' + str(abs(np.mean(np.array(self.diffs)))))
+        print('Std Dev = ' + str(np.std(np.array(self.diffs))))   
+        
+    def graph_residual_error_per_species(self, specie):
+        if specie not in self.species:
+            raise Exception('not a valid specie')
+
+        if len(self.corrections) == 0:
+            self.compute_corrections()
+
+        abs_errors = [abs(i) for i in (self.diffs - np.dot(self.coeff_mat, self.corrections))]
+        labels_species = self.names.copy()
+        diffs_cpy = self.diffs.copy()
+        num = len(labels_species)
+        
+        if specie == 'oxide' or specie == 'peroxide' or specie == 'superoxide':
+            if specie == 'oxide':
+                compounds = self.oxides
+            elif specie == 'peroxide':
+                compounds = self.peroxides
+            else:
+                compounds = self.superoxides
+            for i in range(num):
+                if labels_species[num-i-1] not in compounds:
+                    del labels_species[num-i-1]
+                    del abs_errors[num-i-1]
+                    del diffs_cpy[num-i-1]  
+        else:
+            for i in range(num):
+                if not Composition(labels_species[num-i-1])[specie]:
+                    del labels_species[num-i-1]
+                    del abs_errors[num-i-1]
+                    del diffs_cpy[num-i-1]        
+        abs_errors, labels_species, diffs_cpy = (list(t) for t in zip(*sorted(zip(abs_errors, labels_species, diffs_cpy)))) #sort by error
+        indices = [i for i in range(len(diffs_cpy))]
+        if len(indices) > 20:
+            plt.figure(figsize=(20, 10))
+        plt.scatter(indices, abs_errors)
+        texts = []
+        for i, txt in enumerate(labels_species):
+            texts.append(plt.text(indices[i], abs_errors[i], txt, fontsize = 12))
+        adjust_text(texts)
+        plt.show()
+        print('Residual Error:')
+        print('Median = ' + str(np.median(np.array(abs_errors))))
+        print('Mean = ' + str(np.mean(np.array(abs_errors))))
+        print('Std Dev = ' + str(np.std(np.array(abs_errors))))
+        print('Original Error:')
+        print('Median = ' + str(abs(np.median(np.array(diffs_cpy)))))
+        print('Mean = ' + str(abs(np.mean(np.array(diffs_cpy)))))
+        print('Std Dev = ' + str(np.std(np.array(diffs_cpy)))) 
+
+        print(labels_species) 
+
+    def make_yaml(self, name='MP'):
+        """
+        create the _name_Compatibility.yaml that stores corrections as well as _name_CompatibilityErrors.yaml for correction errors
+        """
+
+        if len(self.corrections) == 0:
+            self.compute_corrections()
+
+        aqueous = OrderedDict()
```

Comment:
Add a comment mentioning these come from the old YAML file and are not yet auto-generated by this class.